### PR TITLE
bpo-41902: Add fast path for long_div if the divisor is one

### PR DESCRIPTION
**Pull request by corona10**
https://bugs.python.org/issue41902

### CODE REVIEW

**Comment by vstinner** [comment]
Can you please run benchmarks? For example, benchmark:
* x//1
* x//2
* x//y where y is made of more than 1 digit

For multiple values of x:

* single digit
* two digits

File: Lib/test/test_long.py [hunk 1]
```diff
@@ -204,7 +204,10 @@ def test_division(self):
         self.check_division(710031681576388032, 26769404391308)
         self.check_division(1933622614268221, 30212853348836)
 
-
+    @support.cpython_only
+    def test_division_with_one(self):
+        for n in range(10000000):
+            self.assertEqual(n, n // 1)
```
Comment (vstinner) [inline Lib/test/test_long.py:210]: I suggest to test assertIs().

File: Lib/test/test_long.py [hunk 2]
```diff
@@ -204,7 +204,10 @@ def test_division(self):
         self.check_division(710031681576388032, 26769404391308)
         self.check_division(1933622614268221, 30212853348836)
 
-
+    @support.cpython_only
+    def test_division_with_one(self):
+        for n in range(10000000):
```
Comment (vstinner) [inline Lib/test/test_long.py:209]: Well, I don't think that it's worth it to test 10000000 numbers. Just test a bunch of various numbers made of a single digit. Like:

```
(-MASK, -10, -5, 0, 1, 2, 42, MASK)
```

File: Lib/test/test_long.py [hunk 3]
```diff
@@ -204,7 +204,10 @@ def test_division(self):
         self.check_division(710031681576388032, 26769404391308)
         self.check_division(1933622614268221, 30212853348836)
 
-
+    @support.cpython_only
+    def test_division_with_one(self):
```
Comment (vstinner) [inline Lib/test/test_long.py:208]: If you test assertIs(), please add a comment mentioning bpo-41902 and explain that it's a micro-optimization which returns x for x//1 if x is an int made of a single digit.

File: Lib/test/test_long.py [hunk 4]
```diff
@@ -204,7 +204,10 @@ def test_division(self):
...
+        for n in range(10000000):
+            self.assertEqual(n, n // 1)
```
Comment (vstinner) [inline Lib/test/test_long.py:211]: Test also that assertEqual(type(x//1), int) where x is an int subclass.

File: Objects/longobject.c [hunk 5]
```diff
@@ -3695,6 +3695,10 @@ static PyObject *
 long_div(PyObject *a, PyObject *b)
 {
     PyLongObject *div;
+    if (b == _PyLong_One && PyLong_CheckExact(a)) {
```
Comment (pablogsal) [inline Objects/longobject.c:3698]: This branch *may* impact the normal path, we certainly should benchmark this with PGO and cpu isolation.

**Pull request closed without merging**